### ISSUE TITLE
sqlite3.OperationalError: unable to open database file, final result: no image browser tab

### ISSUE THREAD

**Issue by cibernicola** [AlUlkesh/stable-diffusion-webui-images-browser#38]
`Error executing callback ui_tabs_callback for L:\stable-diffusion-webui\extensions\stable-diffusion-webui-images-browser\scripts\image_browser.py
Traceback (most recent call last):
  File "L:\stable-diffusion-webui\modules\script_callbacks.py", line 106, in ui_tabs_callback
    res += c.callback() or []
  File "L:\stable-diffusion-webui\extensions\stable-diffusion-webui-images-browser\scripts\image_browser.py", line 806, in on_ui_tabs
    create_tab(tab)
  File "L:\stable-diffusion-webui\extensions\stable-diffusion-webui-images-browser\scripts\image_browser.py", line 562, in create_tab
    exif_cache = wib_db.load_exif_data(exif_cache)
  File "L:\stable-diffusion-webui\extensions\stable-diffusion-webui-images-browser\scripts\wib\wib_db.py", line 305, in load_exif_data
    cursor.execute('''
sqlite3.OperationalError: unable to open database file`

Extension version: just updated
OS: Win 10
WebUI version: latest
¿Any suggestion?

**Comment by cibernicola** [comment]
a) Didn't check it, must it be created manually?
b) Nope
c) 133GB remaining
updating soon :D

**Comment by cibernicola** [comment]
Hums, nope, not sure what's goin' on, 
Restarting WebUI:
`Error executing callback ui_tabs_callback for L:\stable-diffusion-webui\extensions\stable-diffusion-webui-images-browser\scripts\image_browser.py
Traceback (most recent call last):
  File "L:\stable-diffusion-webui\modules\script_callbacks.py", line 106, in ui_tabs_callback
    res += c.callback() or []
  File "L:\stable-diffusion-webui\extensions\stable-diffusion-webui-images-browser\scripts\image_browser.py", line 806, in on_ui_tabs
    create_tab(tab)
  File "L:\stable-diffusion-webui\extensions\stable-diffusion-webui-images-browser\scripts\image_browser.py", line 562, in create_tab
    exif_cache = wib_db.load_exif_data(exif_cache)
  File "L:\stable-diffusion-webui\extensions\stable-diffusion-webui-images-browser\scripts\wib\wib_db.py", line 305, in load_exif_data
    cursor.execute('''
sqlite3.OperationalError: unable to open database file`

meanwhile, the database file: 
![image](https://user-images.githubusercontent.com/4579387/219788755-edf0d230-2e9e-4a97-b1c4-c95d7c7fe395.png)

**Comment by AlUlkesh** [comment]
This is very strange. I just checked and if the database was not there it would be created. And your last example shows the file.

Is there anything running on your machine that might lock the file? Backup/syncing processes? Anti-Virus?

Is this a normal or some other drive (network, etc.)?

**Comment by cibernicola** [comment]
Hi:
A/V: Malwarebytes 
Local Hard Disk
Processes having this file locked: (as far as I can check) 0
![image](https://user-images.githubusercontent.com/4579387/219939889-1fd34150-aa7c-4cdb-ad1d-6af9c1962441.png)

The behavior has been weird: I deleted the extension directory, restarted WebUI, it worked. I used it, etc..., everything was fine until I restarted it again :S

**Comment by AlUlkesh** [comment]
I'm also kind of at a loss here. So weird.

Can you use something like _DB Browser for SQLite_ to check if the database got perhaps corrupted somehow?

**Comment by cibernicola** [comment]
It seems to be fine:
![image](https://user-images.githubusercontent.com/4579387/219941433-6757b45d-ab9f-4b93-9dbb-744bdb860158.png)

No clue whats goin' on :S

**Comment by cibernicola** [comment]
Behavioral pattern:
- If the database does NOT exist, it is created and the extension works correctly.
- If the database exists, when initializing WebUI, the extension "cannot" open the SQLite file.
-If the database file is created manually the extension cannot open it either.

Which "tells me" that at the moment when the extension generates data and stores it in the SQLite file, "something happens". I don't know if it is related to permissions or something extra :S

**Comment by michikora** [comment]
I had the same problem, but I may have found a possible reason for it.

After I completely cleared all the thumbnails from temp folder, image browser worked again, even after I restarted webui.
I'm not sure exactly which files are causing the problem, but they contain two files that I think could be the cause.
- File name is too long (longer than the default max_path)
- File name contains a comma
Sorry, I don't have time to test which file it was right now, but deleting them did fix the problem.

By the way, if you haven't set the path to save thumbnails, they should be in %temp%(C:\Users\<Username>\AppData\Local\Temp)

**Comment by michikora** [comment]
Ummm...
I just did a quick try to put back the files that may have caused error, but everything seems to be working fine so far.
So weird.

**Comment by cibernicola** [comment]
Bingo!
It is a case that mixes several failures, among them and perhaps the most important, the one caused by the user.
In the startup file I had changed the temporary directories, but for some reason, they were wrong (they were not accessible). As soon as I corrected them, everything works again :)

PS: The "several errors" I say it because it seems that "nobody" checks that the paths are accessible (?)


Restarting WebUI several times, and, finally, working fine.

Thanks :)
